libfwupd/fwupd-client-sync.c

```diff
@@ -673,6 +673,112 @@ fwupd_client_unlock(FwupdClient *self,
 	}
 	return TRUE;
 }
+
+static void
+fwupd_client_inhibit_cb(GObject *source, GAsyncResult *res, gpointer user_data)
+{
+	FwupdClientHelper *helper = (FwupdClientHelper *)user_data;
+	helper->str = fwupd_client_inhibit_finish(FWUPD_CLIENT(source), res, &helper->error);
+	g_main_loop_quit(helper->loop);
+}
+
+/**
+ * fwupd_client_inhibit:
+ * @self: a #FwupdClient
+ * @reason: (not nullable): the inhibit reason, e.g. `user active`
+ * @cancellable: (nullable): optional #GCancellable
+ * @error: (nullable): optional return location for an error
+ *
+ * Marks all devices as unavailable for update. Update is only available if there is no other
+ * inhibit imposed by other applications or by the system (e.g. low power state).
+ *
+ * The same application can inhibit a device multiple times.
+ *
+ * Returns: (transfer full): a string, or %NULL for failure
```

Comment:
So this presumably returns a string that you use to undo inhibit, right?

You'll probably want to document it.

Reply:
Fixed.